src/Components/ServiceScene/JSONPanel/LogsJsonComponent.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { useCallback, useRef } from 'react';
+
+import { css } from '@emotion/css';
+
+import { FieldType, GrafanaTheme2 } from '@grafana/data';
+import { AdHocFilterWithLabels, SceneComponentProps, sceneGraph } from '@grafana/scenes';
+import { Alert, Badge, PanelChrome, useStyles2 } from '@grafana/ui';
+
+import { isLogLineField } from '../../../services/fields';
+import { getLogsHighlightStyles } from '../../../services/highlight';
+import {
+  setJsonHighlightVisibility,
+  setJsonLabelsVisibility,
+  setJsonMetadataVisibility,
+} from '../../../services/store';
+import { getFieldsVariable, getJsonFieldsVariable, getLineFiltersVariable } from '../../../services/variableGetters';
+import { LogsPanelHeaderActions } from '../../Table/LogsHeaderActions';
+import { NoMatchingLabelsScene } from '../Breakdowns/NoMatchingLabelsScene';
+import LabelRenderer from '../JSONPanel/LabelRenderer';
+import ValueRenderer from '../JSONPanel/ValueRenderer';
+import { LogListControls } from '../LogListControls';
+import { LogsJsonScene } from '../LogsJsonScene';
+import { LogsListScene } from '../LogsListScene';
+import { getLogsPanelFrame } from '../ServiceScene';
+import GetItemString from './GetItemString';
+import { JSONTree } from '@gtk-grafana/react-json-tree';
+
+export default function LogsJsonComponent({ model }: SceneComponentProps<LogsJsonScene>) {
+  const {
+    data,
+    emptyScene,
+    hasJsonFields,
+    jsonFiltersSupported,
+    menu,
+    showHighlight,
+    showLabels,
+    showMetadata,
+    sortOrder,
+  } = model.useState();
+  const $data = sceneGraph.getData(model);
+  // Rerender on data change
+  $data.useState();
+  const logsListScene = sceneGraph.getAncestor(model, LogsListScene);
+  const { visualizationType } = logsListScene.useState();
+  const styles = useStyles2((t) => getStyles(t, showHighlight));
+
+  const fieldsVar = getFieldsVariable(model);
+  const jsonVar = getJsonFieldsVariable(model);
+
+  // If we have a line format variable, we are drilled down into a nested node
+  const dataFrame = getLogsPanelFrame(data);
```

Comment:
One big limitation that I see from this approach of highlighting based in variables is that you have zero certainty of what Loki is going to match; we're guessing in the frontend. The second is the added complexity of parsing line filters for highlighting.

To have matching behaviors between visualizations and highlight what Loki effectively matches, I suggest to use `searchWords` from the data frame, like logs model does. That, in combination with this function https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/text/text.ts#L73 should give you the regex to match.

You can find the search words as meta information in the frame coming from the data source:

![imagen](https://github.com/user-attachments/assets/e895e62c-1a02-4b00-8a64-bd7f90d24b24)

Reply:
So yeah, this is being added to the frontend dataframe in backendResultTransformer by parsing the Loki expression with lezer from the frontend.

Reply:
Unfortunate. That's really no different from evaluating the filter variables. All good then, thanks for checking!